lib/galaxy/datatypes/phylip.py

```diff
@@ -0,0 +1,77 @@
+"""
+Created on January. 05, 2018
+
+@authors: Kenzo-Hugo Hillion and Fabien Mareuil, Institut Pasteur, Paris
+@contacts: kehillio@pasteur.fr and fabien.mareuil@pasteur.fr
+@project: galaxy
+@githuborganization: C3BI
+Phylip datatype sniffer
+"""
+
+from metadata import MetadataElement
+
+from galaxy import util
+from galaxy.datatypes.data import get_file_peek, Text
+from galaxy.datatypes.sniff import get_headers
+from galaxy.util import nice_size
+
+
+class Phylip(Text):
+    """Phylip format stores a multiple sequence alignment"""
+    edam_data = "data_0863"
+    edam_format = "format_1997"
+    file_ext = "phylip"
+
+    """Add metadata elements"""
+    MetadataElement(name="sequences", default=0, desc="Number of sequences", readonly=True,
+                    visible=False, optional=True, no_value=0)
+
+    def set_meta(self, dataset, **kwd):
+        """
+        Set the number of sequences and the number of data lines in dataset.
+        """
+        dataset.metadata.data_lines = self.count_data_lines(dataset)
+        try:
+            dataset.metadata.sequences = int(get_headers(dataset.file_name, '\t', count=1)[0][0].split()[0])
```

Comment:
You can use `int(get_headers(dataset.file_name, '\t', count=1)[0][0])`, `get_headers` return a list of lists with the outer list corresponding to lines and the inner list fields as split by sep.

```
In [7]: int(get_headers('/Users/mvandenb/src/galaxy/t', '\t', count=1)[0][0])
Out[7]: 12
```

Reply:
This is what I intended to do first. But then I realized the header of the PHYLIP format was tricky and somehow separated by spaces. I am not sure the number of spaces is always the same so I am not able to be 100% to refer to the right element of the list (I am not sure to be 100% clear).

Reply:
That shouldn't matter:
```
In [5]: int('      \t  12    ')
Out[5]: 12
```

Reply:
It is just that I am not sure how to properly split the header. In my example it is separated by 3 spaces but there is no rule about it. [Here](http://scikit-bio.org/docs/0.2.3/generated/skbio.io.phylip.html) it says that "*The header consists of a single line describing the dimensions of the alignment. It must be the first line in the file. The header consists of optional spaces, followed by two positive integers (n and m) separated by one or more spaces.*"

Reply:
Alright, in that case I'd go back to `int(open(dataset.file_name).readline().split()[0])` or similar. We don't gain anything by using `get_headers`, sorry about that.